Refactor : Stop using total records in front end

keithCuniah opened the pull request:
# Description
Since the `total number records` is not include anymore in the call to fetch records (See #2917) , the value have to be removed from the feedback task.

List of impacts by components/model :

1/ FeedbackPagination.component.vue :
the total_records is used to compute the number of total page
it's also used to render the pagination component (if no total_records, the component is not rendered)

2/ CenterFeedbackTask.content.vue :
when total_records change, we reinitiate the flag areResponsesUntouched to true (this help us to show are not a toast on change route or on change filter)
also used to know in which page the user is in order to allow to go to next record or not on submit/discard

3/ RecordFeedbackTaskAndQuestionnaire.content.vue :
it's where the total_records is fetch and insert in the orm
it's used as a flag to show or not a message in case there is no record

4/ FeedbackTask.model.js && feedbackTask.queries.js :
where there is the attribute total_record && queries to update/get the value from the orm

=> When the total record will be removed from the api, we have to clean the front orm and adapt the 3 components to get the toal record from the metrics instead of the previous way


**Type of change**

- Refactor (change restructuring the codebase without changing functionality)

**How Has This Been Tested**

- Feedback task only

frascuchon committed:
Remove the use of totalRecords and all related logic

This changes implies some code refatoring which is already done

frascuchon commented on frontend/components/feedback-task/footer/Pagination.component.vue:
```diff
@@ -63,59 +40,28 @@ export default {
     isPageAvailable() {
       return this.pageFromRoute <= this.totalPages;
     },
+    // TODO: Move to the parent
```
Do we still need this computed value? If not, we should remove it

gabrielmbmb committed:
Remove `total` field from output payload of listing records endpoints (#2969)

# Description

I've removed the `total` field from the output payload of the endpoints
listing records in API v1 (`GET /api/v1/datasets/{dataset_id}/records`
and `GET /api/v1/me/datasets/{dataset_id}/records`), and also in the
Python client (`FeedbackRecordsModel`).

Closes #2917 (backend side only)

**Type of change**

(Please delete options that are not relevant. Remember to title the PR
according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing
functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing
functionality)
- [ ] Improvement (change adding some improvement to an existing
functionality)
- [ ] Documentation update

**How Has This Been Tested**

All tests in `tests/server/api/v1/test_datasets.py` have passed and also
the ones from `tests/client/{test_feedback.py, sdk/v1/datasets}`

**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my
feature works
- [ ] I have added relevant notes to the CHANGELOG.md file (See
https://keepachangelog.com/)

frascuchon committed:
Fixing error with empty data
